src/cfnlint/rules/resources/codepipeline/CodepipelineStageActions.py

```diff
@@ -0,0 +1,228 @@
+                    b=max_
+                )
+                matches.append(RuleMatch(
+                    path + [artifact_type],
+                    message
+                ))
+        else:
+            if artifact_count != constraints[constraint_key]:
+                message = (
+                    'Action "{action}" declares {number} {artifact_type} which is less than '
+                    'expected [{a}].'
+                ).format(
+                    action=action['Name'],
+                    number=artifact_count,
+                    artifact_type=artifact_type,
+                    a=constraints[constraint_key]
+                )
+                matches.append(RuleMatch(
+                    path + [artifact_type],
+                    message
+                ))
+
+        return matches
+
+    def check_owner(self, action, path):
+        """Check that action type owner is valid."""
+        matches = []
+
+        owner = action.get('ActionTypeId', {}).get('Owner')
+        if owner not in self.VALID_OWNER_STRINGS:
+            message = (
+                'For all currently supported action types, the only valid owner '
+                'strings are {owners}'
+            ).format(
+                owners=', '.join(list(self.VALID_OWNER_STRINGS))
+            )
+            matches.append(RuleMatch(
+                path + ['ActionTypeId', 'Owner'],
+                message
+            ))
+
+        return matches
+
+    def check_version(self, action, path):
+        """Check that action type version is valid."""
+        matches = []
+
+        if action.get('ActionTypeId', {}).get('Version') != 1:
+            message = 'For all currently supported action types, the only valid version is 1.'
+            matches.append(RuleMatch(
+                path + ['ActionTypeId', 'Version'],
+                message
+            ))
+
+        return matches
+
+    def check_names_unique(self, action, path, action_names):
+        """Check that action names are unique."""
+        matches = []
+
+        if action.get('Name') in action_names:
+            message = 'All action names within a stage must be unique. ({name})'.format(
+                name=action.get('Name')
+            )
+            matches.append(RuleMatch(path + ['Name'], message))
+        action_names.add(action.get('Name'))
+
+        return matches
+
+    def match(self, cfn):
+        """Check that stage actions are set up properly."""
+        matches = list()
+
+        resources = cfn.get_resource_properties(['AWS::CodePipeline::Pipeline'])
+        for resource in resources:
+            path = resource['Path']
+            properties = resource['Value']
+
+            for sidx, stage in enumerate(properties['Stages']):
+                stage_path = path + ['Stages', sidx]
+                action_names = set()
+                for aidx, action in enumerate(stage['Actions']):
+                    action_path = stage_path + ['Actions', aidx]
+
```

Comment:
could do 
```
try:
  matches.extend(self.check_names_unique(action, action_path, action_names))
  matches.extend(self.check_version(action, action_path))
  matches.extend(self.check_owner(action, action_path))
  matches.extend(self.check_artifact_counts(action, 'InputArtifacts', action_path))
  matches.extend(self.check_artifact_counts(action, 'OutputArtifacts', action_path))
except AttributeError as err:
  self.logger.debug('Got Attribute Error. Should have been caught by generic linting. '
                                 'Ignoring the error here: %s', err)
```

This would result in template.
```
Stages:
        - Name: Source
          Actions:
            - Name: Github
              ActionTypeId:
              - Category: Source
                Owner: ThirdParty
                Provider: GitHub
                Version: 1
              OutputArtifacts:
                - Name: MyApp
              Configuration:
                Owner: awslabs
                Repo: cfn-python-lint
                PollForSourceChanges: true
                Branch: master
                OAuthToken: 'secret-token'
```
Getting Error (instead of crashing)
```
E3002 Expecting an object at Resources/TestPipeline/Properties/Stages/0/Actions/0/ActionTypeId
test/templates/bad/resources_codepipeline_stages_one_stage.yaml:17:15
```